unicodetools/org/unicode/tools/emoji/ProposalData.java

```diff
@@ -407,14 +441,22 @@ private static String getName(String emoji) {
 	}
     }
 
-    static ProposalData SINGLETON = new ProposalData();
+    private static final class Helper {
+	static ProposalData SINGLETON = new ProposalData();
+    }
 
     public static ProposalData getInstance() {
-	return SINGLETON;
+	return Helper.SINGLETON;
     }
 
     public static void chart(String dir, String filename) {
 	ProposalData instance = getInstance();
+	
+	try (PrintWriter pw = FileUtilities.openUTF8Writer(dir, "emoji-proposals.txt")) {
```

Comment:
We used to only be generating emoji-proposals on demand. Now it is always generated by GenerateEmoji.